src/coreclr/vm/methodtablebuilder.cpp

```diff
@@ -11509,44 +11575,51 @@ VOID MethodTableBuilder::HandleGCForValueClasses(MethodTable ** pByValueClassCac
         {
             if (pFieldDescList[i].IsByValue())
             {
-                MethodTable *pByValueMT = pByValueClassCache[i];
+                MethodTable* pByValueMT = pByValueClassCache[i];
 
                 if (pByValueMT->ContainsPointers())
                 {
                     // Offset of the by value class in the class we are building, does NOT include Object
                     DWORD       dwCurrentOffset = pFieldDescList[i].GetOffset_NoLogging();
+                    DWORD       dwElementSize = pByValueMT->GetBaseSize() - OBJECT_BASESIZE;
 
-                    // The by value class may have more than one pointer series
-                    CGCDescSeries * pByValueSeries = CGCDesc::GetCGCDescFromMT(pByValueMT)->GetLowestSeries();
-                    SIZE_T dwNumByValueSeries = CGCDesc::GetCGCDescFromMT(pByValueMT)->GetNumSeries();
-
-                    for (SIZE_T j = 0; j < dwNumByValueSeries; j++)
+                    for (DWORD r = 0; r < repeat; r++)
```

Comment:
If the value is fully covered by one series, we should only emit one series for the repeated value.

Reply:
Like in arrays? I think that pattern requires that the element count is stored in the instance.

Reply:
Except for side-by-side objects. That we can express as one combined sequence.

Reply:
That seems like can be handled in more general terms - including regular structs. 
If it is an all-objects instance (recursively) and it is not an explicit layout, we can put in just one serie for the entire size of the instance.

Reply:
I've looked at this from implementation point. Folding GC series for side-by-side objects from separate struct fields is certainly doable, but a bit involved, especially on the CoreClr side. 

I think it is better if this is handled as a separate PR as it is an optimization and we may need to iterate a bit on it. There is also some risk there will be bugs or regressions.

Reply:
Logged an issue for this - https://github.com/dotnet/runtime/issues/83111